src/main/java/com/epam/lab/paymentsystem/controller/OperationController.java

```diff
@@ -0,0 +1,137 @@
+package com.epam.lab.paymentsystem.controller;
+
+import com.epam.lab.paymentsystem.entities.Card;
+import com.epam.lab.paymentsystem.entities.Operation;
+import com.epam.lab.paymentsystem.service.CardService;
+import com.epam.lab.paymentsystem.service.OperationService;
+import java.time.LocalDateTime;
+import java.util.List;
+import javax.transaction.Transactional;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+/**
+ * Controller for Payment operation,
+ * processes the specified URL and redirects
+ * the request to the specified page.
+ *
+ * @author unascribed
+ * @since 0.0.1
+ */
+@Controller
+public class OperationController {
+
+  private static final Logger LOGGER = LogManager.getLogger(OperationController.class);
+  private static final String HISTORY_PAGE = "history";
+
+  @Autowired
+  private OperationService operationService;
+
+  @Autowired
+  private CardService cardService;
+
+
+  /**
+   * Get method for operation page.
+   *
+   * @param model model
+   * @return String
+   */
+  @GetMapping(value = "/operation")
+  public String getOperationPage(Model model) {
+    String operationPage = "operation";
+    model.addAttribute("srcCardList", cardService.getAllCardsByCurrentUser());
+    model.addAttribute("dstCardList", cardService.getAllCards());
+    return operationPage;
+  }
+
+  /**
+   * Payment operation .
+   *
+   * @param srcId  Long, source card.
+   * @param dstId  Long, destination card.
+   * @param amount Long, amount money.
+   * @param model  Model.
+   * @return String
+   */
+  @PostMapping(value = "/operation")
+  @Transactional(rollbackOn = Exception.class)
+  public String paymentOperation(@RequestParam(name = "src") Long srcId,
+                                 @RequestParam(name = "dst") Long dstId,
+                                 @RequestParam(name = "amount") Long amount,
+                                 Model model) {
+    String operationPage = "operation";
+    LocalDateTime date = LocalDateTime.now().withNano(0);
+    Operation operation = new Operation();
```

Comment:
This logic should be in service layer

Reply:
Right remark, now logic is in service  + I made writeHistory method protected and invoked by makePayment method (In case when transaction is successfully done)